cpp/src/randomforest/randomforest.cuh

```diff
@@ -141,6 +141,20 @@ class RandomForest {
            n_streams,
            handle.get_stream_pool_size());
 
+    // allocating memory for quantile structure storing device pointers
+    auto quantiles_array = std::make_shared<rmm::device_uvector<T>>(
+      this->rf_params.tree_params.n_bins * n_cols, handle.get_stream());
+    auto n_uniquebins_array =
+      std::make_shared<rmm::device_uvector<int>>(n_cols, handle.get_stream());
+    // creating quantile structure storing device pointers
+    DT::Quantiles<T, int> quantiles = {quantiles_array->data(), n_uniquebins_array->data()};
+    // computing the quantiles
+    DT::computeQuantiles(
```

Comment:
It's better to just return these from the DT::computeQuantiles function. We can do this easily using shared pointers and structured bindings.

Why is it better to use return values instead of passing output structures as function arguments? https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f21-to-return-multiple-out-values-prefer-returning-a-struct-or-tuple

Notice that your test code becomes a lot less verbose if you do this.

Reply:
so if I allocate memory for the members of `Quantiles` inside the `computeQuantiles()` func and return it, the memory is not accessible outside the function scope. So I resorted to allocating it outside and modifying it's contents in the `computeQuantiles`. In my following commit I have additionally explicitly returned it. Did you have that in mind, or is there another way using pointers to preserve the memory?